### PR TITLE
fix: toast message added for adding report to expense in both modes

### PR DESCRIPTION
**Pull request by suyashpatil78**
### <samp>🤖 Generated by Copilot at 51aad18</samp>

This pull request adds and improves the toast messages that are shown to the user when adding or removing expenses to and from reports in the fyle-mobile-app. It covers the transactions-outbox.service, the AddEditExpensePage, the AddEditMileagePage and the AddEditPerDiemPage components and their respective spec files. It also fixes some formatting and code style issues in the files.

### <samp>🤖 Generated by Copilot at 51aad18</samp>

> _Toast messages show_
> _When expenses change report_
> _`AddEdit` pages - fall_

### <samp>🤖 Generated by Copilot at 51aad18</samp>

*  Add new methods to show toast messages when adding or removing expenses from reports in various components and services ([link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-517c7bbb54736d0373e068f6cfc16f428e2618afa23906d7e093afbf7b34d65dR76-R104), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-517c7bbb54736d0373e068f6cfc16f428e2618afa23906d7e093afbf7b34d65dR440-R453), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bR3446-R3455), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL3778-R3790), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL3786-R3801), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL3793-R3811), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eR1764-R1776), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL2271-R2287), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL2279-R2298), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL2286-R2308), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL2037-R2040), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL2045-R2051), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dL2052-R2061), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-b6e1f6e315ef2edc53799d40c1304deff11e97673e1003986d199937196dda0dR2103-R2115))
*  Add new test cases to verify the logic and functionality of the new methods in the components and services ([link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-fc679012f8cc3980f7afc0f8785b09245d9bb39050ca7c39ca780e05d7c0c930R112-R152), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57R393-R406), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57R945-R946), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57R984-R985), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57R1019-R1020), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-a399a0aa9b21b631441eb6d1767af446a7b50dbe5602e1c8acb44721db6f83e5R629-R630), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-a399a0aa9b21b631441eb6d1767af446a7b50dbe5602e1c8acb44721db6f83e5R674-R675), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-a399a0aa9b21b631441eb6d1767af446a7b50dbe5602e1c8acb44721db6f83e5R720-R721), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-b97456a1337050be58f0238bfa7892f70e10c42a7d58754c5d1528a1a0952d7fR717-R730), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eR345-R346), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL389-R400), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eR443-R444), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eR827-R840))
*  Import the required dependencies for the new features in the components and services ([link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-fc679012f8cc3980f7afc0f8785b09245d9bb39050ca7c39ca780e05d7c0c930R21-R24), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-517c7bbb54736d0373e068f6cfc16f428e2618afa23906d7e093afbf7b34d65dR21-R24), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L162-R166), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL23))
*  Declare and create the spy objects for the mocked services of MatSnackBar, SnackbarPropertiesService and Router in the transactions-outbox.service.spec.ts file ([link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-fc679012f8cc3980f7afc0f8785b09245d9bb39050ca7c39ca780e05d7c0c930R40-R42), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-fc679012f8cc3980f7afc0f8785b09245d9bb39050ca7c39ca780e05d7c0c930L44-R55))
*  Provide and inject the mocked services of MatSnackBar, SnackbarPropertiesService and Router in the transactions-outbox.service.spec.ts file ([link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-fc679012f8cc3980f7afc0f8785b09245d9bb39050ca7c39ca780e05d7c0c930R69-R71), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-fc679012f8cc3980f7afc0f8785b09245d9bb39050ca7c39ca780e05d7c0c930R85-R87))
*  Remove the trailing commas from the last elements of the objects and arrays in the spec files to follow the code style guidelines ([link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L349-R349), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L451-R465), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L469-R483), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L485-R499), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L507-R521), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L515-R529), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L535-R549), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L557-R571), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L565-R579), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L582-R596), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L612-R626), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L629-R651), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L660-R674), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L678-R705), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L714-R728), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L734-R761), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L772-R786), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L786-R808), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L819-R833), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L833-R855), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L1014-R1039), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L1038-R1063), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L1063-R1088), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L1091-R1116), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L1102-R1127), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L1108-R1133), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L1126-R1151), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L1153-R1178), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57L1164-R1189), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL226-R226), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL249-R249), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL290-R292), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL318-R320), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL363-R367), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL409-R417), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL452-R462), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL486-R496), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL492-R502), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL511-R521), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL545-R555), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL551-R561), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL568-R578), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL602-R612), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL608-R618), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL625-R635), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL658-R668), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL680-R690), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL713-R723), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL719-R729), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL736-R746), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eL776-R786))
*  Remove the noop function from the addEntry method of the transactionOutboxService in the add-edit-expense.page.ts file as it is no longer needed ([link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL4124-R4146))
*  Spy on the showAddToReportSuccessToast and showReportRemovedToast methods of the components and prevent their actual execution in the spec files ([link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-f43cd5c38b66131b82307192d0e690768498b3d43e4b378f36d321f42487ab57R910-R914), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-a399a0aa9b21b631441eb6d1767af446a7b50dbe5602e1c8acb44721db6f83e5R576-R577), [link](https://github.com/fylein/fyle-mobile-app/pull/2536/files?diff=unified&w=0#diff-9fc3d987464b78b0c9c205e0855313051027757354ed26bcb90efd102618948eR281-R282))

## Clickup
https://app.clickup.com/t/86791a5cz

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts [hunk 1]
```diff
@@ -390,6 +390,20 @@ export function TestCases4(getTestBed) {
       }));
     });
 
+    it('showReportRemovedToast(): should show expense removed from report toast', () => {
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts:393]: change here

File: src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts [hunk 2]
```diff
@@ -893,6 +907,11 @@ export function TestCases4(getTestBed) {
     });
 
     describe('editExpense():', () => {
+      beforeEach(() => {
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts:910]: change here

File: src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts [hunk 3]
```diff
@@ -923,6 +942,8 @@ export function TestCases4(getTestBed) {
           expect(transactionService.upsert).toHaveBeenCalledOnceWith(newUnflattenedTxn.tx);
           expect(transactionService.getETxnUnflattened).toHaveBeenCalledOnceWith('txNVtsqF8Siq');
           expect(reportService.addTransactions).toHaveBeenCalledOnceWith('rprAfNrce73O', ['tx3qHxFNgRcZ']);
+          expect(component.showAddToReportSuccessToast).toHaveBeenCalledTimes(1);
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts:945]: change

File: src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts [hunk 4]
```diff
@@ -960,6 +981,8 @@ export function TestCases4(getTestBed) {
           expect(transactionService.getETxnUnflattened).toHaveBeenCalledOnceWith('txbO4Xaj4N53');
           expect(reportService.addTransactions).toHaveBeenCalledOnceWith('rprAfNrce73O', ['txbO4Xaj4N53']);
           expect(reportService.removeTransaction).toHaveBeenCalledOnceWith('rpGpzBpAxtSn', 'txbO4Xaj4N53');
+          expect(component.showAddToReportSuccessToast).toHaveBeenCalledTimes(1);
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts:984]: change

File: src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts [hunk 5]
```diff
@@ -993,6 +1016,8 @@ export function TestCases4(getTestBed) {
           expect(authService.getEou).toHaveBeenCalledTimes(1);
           expect(component.trackEditExpense).toHaveBeenCalledOnceWith(unflattenedTxnDataWithReportID2);
           expect(reportService.removeTransaction).toHaveBeenCalledOnceWith('rpGpzBpAxtSn', 'txbO4Xaj4N53');
+          expect(component.showAddToReportSuccessToast).not.toHaveBeenCalled();
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-expense/add-edit-expense-4.spec.ts:1019]: change

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem-4.page.spec.ts [hunk 6]
```diff
@@ -278,6 +278,8 @@ export function TestCases4(getTestBed) {
       const etxn$ = of({ tx: unflattenedTxnData.tx, ou: unflattenedTxnData.ou, dataUrls: [] });
       const customFields$ = of(txnCustomProperties4);
       beforeEach(() => {
+        spyOn(component, 'showAddToReportSuccessToast');
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem-4.page.spec.ts:281]: change

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem-4.page.spec.ts [hunk 7]
```diff
@@ -814,6 +824,20 @@ export function TestCases4(getTestBed) {
       });
     });
 
+    it('showReportRemovedToast(): should show toast message on removing expense from report', () => {
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem-4.page.spec.ts:825]: change here

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem-4.page.spec.ts [hunk 8]
```diff
@@ -432,6 +438,8 @@ export function TestCases4(getTestBed) {
             expect(transactionService.getETxnUnflattened).toHaveBeenCalledOnceWith(unflattenedTxnData.tx.id);
             expect(reportService.addTransactions).not.toHaveBeenCalled();
             expect(reportService.removeTransaction).toHaveBeenCalledOnceWith('rp8eUleN29dc', 'tx3qHxFNgRcZ');
+            expect(component.showAddToReportSuccessToast).not.toHaveBeenCalled();
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem-4.page.spec.ts:441]: change

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem-4.page.spec.ts [hunk 9]
```diff
@@ -340,6 +342,8 @@ export function TestCases4(getTestBed) {
             expect(transactionService.upsert).toHaveBeenCalledOnceWith(unflattenedTxnData.tx);
             expect(transactionService.getETxnUnflattened).toHaveBeenCalledOnceWith(unflattenedTxnData.tx.id);
             expect(reportService.addTransactions).toHaveBeenCalledOnceWith('rp5eUkeNm9wB', ['tx3qHxFNgRcZ']);
+            expect(component.showAddToReportSuccessToast).toHaveBeenCalledTimes(1);
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem-4.page.spec.ts:345]: change

File: src/app/core/services/transactions-outbox.service.ts [hunk 10]
```diff
@@ -401,6 +454,7 @@ export class TransactionsOutboxService {
                   .addTransactions(reportId, txnIds)
                   .toPromise()
                   .then(() => {
+                    this.formatToastMessage(reportId, entry);
```
Comment (Julias0) [inline src/app/core/services/transactions-outbox.service.ts:457]: This wont work since the sync may happen a long time in the future

**Review by Julias0 — CHANGES_REQUESTED** [review]
Added comment

File: src/app/core/services/transactions-outbox.service.ts (hunk 10 above)
Comment (suyashpatil78) [reply src/app/core/services/transactions-outbox.service.ts:457]: Adding timer of 5 minutes after which toast will not appear

**Pull request closed without merging**